### ISSUE TITLE
p11tool --list-token-urls not listing infineon token

### ISSUE THREAD

**Issue by bhaskarmr** [tpm2-software/tpm2-pkcs11#765]
Hi,
I just followed the following url to setup TPM2.0 with aws green grass but unable to find the tokens for infineon.
see attached screenshot. not sure what went wrong. Tpm2.0 module is attahced to pi and i can see them with "ls /dev/tpm*"
pls help

![image](https://user-images.githubusercontent.com/14790892/162742113-13d5b78b-d1df-4853-8447-44d8268db2ef.png)

**Comment by williamcroberts** [comment]
My guess is it's not finding the tpm2-pkcs11 library. What does this give you:
```bash
strace p11tool --list-tokens 2>&1 | grep tpm2-pkcs11
```

Did you configure p11kit?

**Comment by bhaskarmr** [comment]
Thanks for the response. The above command "strace p11tool --list-tokens 2>&1 | grep tpm2-pkcs11" doesn't output any.
I didnt configure p11kit, i just followed the url [](https://github.com/Infineon/amazon-greengrass-hsi-optiga-tpm ). Pls let me know the steps/commands to configure p11kit.

**Comment by williamcroberts** [comment]
https://github.com/tpm2-software/tpm2-pkcs11/blob/master/docs/P11.md

**Comment by bhaskarmr** [comment]
I did this in fact using the previous URL that I shared has the steps to configure p11kit.  The same results even after re-do this.
List module always gives PKCS trust module and not the device 
![image](https://user-images.githubusercontent.com/14790892/162930167-73777857-a2eb-4230-8b5d-60e0b5ce0ab3.png)

**Comment by williamcroberts** [comment]
Their some differences depending on the version for how to configure it, what's the output of (add -v to your command):
```bash
p11-kit list-modules -v
```

**Comment by williamcroberts** [comment]
Closing stale.

**Comment by VVARELA15** [comment]
I am having the same issue. I noticed there wasn't really a conclusion to this. Was there a way to make the p11-tool see my infineon tpm?

**Comment by williamcroberts** [comment]
> I am having the same issue. I noticed there wasn't really a conclusion to this. Was there a way to make the p11-tool see my infineon tpm?

Does p11 see the pkcs11 provider? Up to that step is all p11kit configuration.

I've never seen a report where folks cannot use it without their being a configuration issue. Make sure the TPM2_PKCS11_STORE is configured, their is a default search algorithm when that env var isn't set. I recommend setting that env var so you know exactly what store you're getting. You should be able to see the store opened in strace output.

**Comment by VVARELA15** [comment]
How can I check that p11 see the pkcs11 provider?

**Comment by williamcroberts** [comment]
> How can I check that p11 see the pkcs11 provider?

For me it's as simple as `p11-kit list-modules` and it outputs:
```
$ p11-kit list-modules
p11-kit-trust: p11-kit-trust.so
    library-description: PKCS#11 Kit Trust Module
    library-manufacturer: PKCS#11 Kit
    library-version: 0.23
    token: System Trust
        manufacturer: PKCS#11 Kit
        model: p11-kit-trust
        serial-number: 1
        hardware-version: 0.23
        flags:
               write-protected
               token-initialized
tpm2-pkcs11: /home/wcrobert/workspace/tpm2-pkcs11/src/.libs/libtpm2_pkcs11.so
    library-description: TPM2.0 Cryptoki
    library-manufacturer: tpm2-software.github.io
    library-version: 0.0
    token: 
        manufacturer: IBM
        model: SW   TPM
        serial-number: 0000000000000000
        hardware-version: 1.62
        firmware-version: 25.35
        flags:
               rng
               login-required
```

**Comment by l-sousa** [comment]
@williamcroberts I'm getting this problem. I installed p11-kit and configured tpm2-pkcs11 again as the instructions tell. 

`
$ export TPM2_PKCS11_STORE=$PWD/

$ tpm2-pkcs11/tools/tpm2_ptool init --path=$TPM2_PKCS11_STORE
action: Created
id: 2

$ tpm2-pkcs11/tools/tpm2_ptool addtoken --pid=1 --label=label --sopin=123456 --userpin=1234 --path=$TPM2_PKCS11_STORE

$ tpm2-pkcs11/tools/tpm2_ptool addkey --algorithm=rsa2048 --label="label" --userpin=1234 --path=$TPM2_PKCS11_STORE
action: add
private:
  CKA_ID: '38376565346432346363616638366234'
public:
  CKA_ID: '38376565346432346363616638366234'

$ p11-kit list-modules
p11-kit-trust: p11-kit-trust.so
    library-description: PKCS#11 Kit Trust Module
    library-manufacturer: PKCS#11 Kit
    library-version: 0.23
    token: System Trust
        manufacturer: PKCS#11 Kit
        model: p11-kit-trust
        serial-number: 1
        hardware-version: 0.23
        flags:
               write-protected
               token-initialized
opensc-pkcs11: opensc-pkcs11.so
    library-description: OpenSC smartcard framework
    library-manufacturer: OpenSC Project
    library-version: 0.21
`

**Comment by l-sousa** [comment]
Ok, so I followed [this tutorial](https://www.mail-archive.com/debian-bugs-dist@lists.debian.org/msg1813916.html).

TLDR:

1. Put [this configuration](https://salsa.debian.org/debian/tpm2-pkcs11/-/blob/01411a3855e39173c6d886455a3d5148f94188d1/misc/p11-kit/tpm2_pkcs11.module) into a new file: /usr/share/p11-kit/modules/tpm2_pkcs11.module 
2. Locate your libtpm2_pkcs11.so
3. go to /usr/lib/aarch64-linux-gnu/pkcs11/ 
4. sudo ln -s /PATH/TO/libtpm2_pkcs11.so

It works now.
